### DISCUSSION TITLE
Install GrandOrgue on Raspberry Pi 4 8GB under RPiOS 64 bit

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#363] [General]
Originally posted 2020-08-29 20:46:41.026000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>

**Comment by larspalo** [comment]
Originally posted 2020-08-29 21:00:29.559000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>    Cannot get GrandOrgue installed using the files in Raspbian_10 subdir all/ and subdir armhf/

Getting following error message:

Following packages have unmet requirements:
grandorgue:armhf :requirements: libwxgtk3.0-gtk3-0v5:armhf (&gt;= 3.0.4+dfsg) but will not be installed
                  requirements: grandorgue-resources:armhf but is not installable 
                  
                  
The aim is to be able to load sample sets larger than 4 GB in GrandOrgue.

Any help is appreciated. Thanks in advance.

**Comment by larspalo** [comment]
Originally posted 2020-08-29 21:56:23.445000 by videophile in GrandOrgue discussion forum on SourceForge<br><br>I use Raspberry Pi 4 with 4Gb RAM and Raspberry Pi OS 32 bits with success, but this doesn&#039;t help. All I can say is that the Pi 4 has enough processing power to run GO nicely, so I hope you find a solution to your problem!

**Reply by larspalo** [reply]
Originally posted 2020-08-29 22:19:36.481000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Thanks for your reaction.&lt;br&gt;
    
      I did the same without any problem. But i would like to get rid of
      the 4GB limitation.
      
      All the best.

**Comment by larspalo** [comment]
Originally posted 2020-09-17 00:09:54.607000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>I wish I could use GO on my raspberry Pi 4, but it&#039;s not loading the .wav files. It actually opens the organ, but my RAM stands like nothing is happening... Can you help me?

**Reply by larspalo** [reply]
Originally posted 2020-09-17 07:24:07.904000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Which GrandOrgue installation file did you use and which organ are you loading?

I had to compile GrandOrgue on my Raspberry Pi 4 with 8 GB Ram to obtain a 64 bit version of GrandOrgue so that i can use the full 8 GB Ram. The version found on in Soundforge for the Raspberry Pi was a 32 bit version limiting the use of Ram to 4 GB.

**Comment by larspalo** [comment]
Originally posted 2020-09-17 16:47:18.405000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>I also have the 8 Gb version, but I could not even make the 32 bit version work properly.
I used the manual instalation for Raspibian, with these codes:

echo &#039;deb http://download.opensuse.org/repositories/home:/e9925248:/grandorgue/Raspbian_10/ /&#039; | sudo tee /etc/apt/sources.list.d/home:e9925248:grandorgue.list

curl -fsSL https://download.opensuse.org/repositories/home:e9925248:grandorgue/Raspbian_10/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/home:e9925248:grandorgue.gpg &gt; /dev/null

sudo apt update

sudo apt install grandorgue

I tryied with Pitea, Azzio and the default organ of GO. I also changed a lot of configuration in the menus, but these all acted tha same as nothing.
Now i&#039;m going to test the package with dpkg command.

**Comment by larspalo** [comment]
Originally posted 2020-09-17 17:22:55.473000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>Well, i did what i said. I also reinstalled GO and its resources, but nothing worked.
Then, i&#039;m going to try to reboot the system with other image of 32 bit Raspberry Pi OS and install again, let&#039;s see what happens...

**Reply by larspalo** [reply]
Originally posted 2020-09-17 18:33:49.102000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Go to this ur:

[https://www.raspberrypi.org/forums/viewtopic.php?f=117&amp;t=275370]

and download the 64 bit image  at

[https://downloads.raspberrypi.org/raspios_arm64/images/raspios_arm64-2020-05-28/2020-05-27-raspios-buster-arm64.zip]

Put this on your SD card.

I attach the GrandOrgue compilation i made with the above RPI OS.
This works here with that OS and gives 8 GB of available memory to GrandOrgue.

I hope this helps.

All the best.<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/7daeed0824/d908/3358/attachment/grandorgue-RPi4-64bit.zip<br>

**Comment by larspalo** [comment]
Originally posted 2020-09-17 20:09:51.055000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>I&#039;m grateful for your solicitude. I&#039;ll send you a feedback when i&#039;m done. Thanks!

**Comment by larspalo** [comment]
Originally posted 2020-09-17 23:59:29.921000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>Your compilation for 64 bits worked on installation, but the same issue I had remains. I can open the organ sample but it doesn&#039;t go to RAM, I don&#039;t know why...
![My raspberry right now:](https://i.imgur.com/QGe1Pj0.png)

**Comment by larspalo** [comment]
Originally posted 2020-09-18 01:06:40.670000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>Finally, I made it work. I only changed the sound output, the latency disired to 16ms and the samples per buffer to 480. Thank you, Eric, a lot to send this compilation for 64 bit.

**Reply by larspalo** [reply]
Originally posted 2020-09-18 07:50:13.174000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Congratulations and enjoy.

I have Piotr Grabowski &#039;s beautiful Giubiasco sample set loaded in GrandOrgue with lossless compression at 44100 Hz and 24 bit resolution. This fits into the available RAM; but a larger sample set like Friesach unfortunately is too big.

All the best.

**Comment by larspalo** [comment]
Originally posted 2020-09-18 18:53:08.187000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>Thank you. Now it&#039;s my turn to help in something...
Here you can download the light version of Friesach that i made editing the reverb samples: 
https://u.pcloud.link/publink/show?code=XZQVD5XZraHzdo2JIrkvPGFlJ5wQU8Ix491k
You consume less than 5GB of RAM and the quality remains the same, but the reverb was reduced a little bit. Most people could not notice difference from the original. Actually even I can&#039;t notice without comparing them side by side.
**Be aware that you must set &quot;Release loading&quot; to &quot;Single release&quot; in &quot;Audio/Midi Settings&quot; to work properly.**

Screen before Friesach light loaded:
![](https://i.imgur.com/VtBDfQH.png)

Screen after Friesach light loaded:
![](https://i.imgur.com/qkfuyz9.png)

**The screenshots are from my Ubuntu Desktop, but I already tested in the Raspberry pi successfully.**

**Reply by larspalo** [reply]
Originally posted 2020-09-20 19:46:57.644000 by videophile in GrandOrgue discussion forum on SourceForge<br><br>Thank you ! I&#039;m using the 4 Gb Rapsberry Pi 4, and I use the Friesach organ. Of course, I had to do some sacrifice: 16 bit, lossless compression, single attack, signle release and remove a few stops. I will try your sample set, maybe I will be able to use more - if not all - stops.

**Reply by larspalo** [reply]
Originally posted 2020-09-20 22:33:17.357000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>Enjoy! Let me,  please,  know if it worked well.  I&#039;m thinking about do this with other Piotr organs and make a post on &quot;general discution&quot; with the links, but i have to know if  the affort  worth it.

**Reply by larspalo** [reply]
Originally posted 2020-09-21 13:06:08.359000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Hello Diego,
    
      Sorry for the late reply, i was absent for a few days.
      
      What a pleasant surprise you sent me.
   
      Thank you very much for this Friesach version.
     
      Thanks to your clear instructions i could load the disposition on
      my RPI4 without any problem.
      
      And as you wrote, there is barely any difference in sound from the
      full version.
     
      Since Friesach is my preferred study organ i don&#039;t need much
      reverb because this could mask playing mistakes.
    
      I guess it requires quite some work to adapt such a large
      disposition.
     
      I read that you consider adapting other Piotr Grabowski samples.
     
      Which ones are you thinking about?
    
      Thanks again for your fine present.
      
      Best regards.
    
      Erik.

**Reply by larspalo** [reply]
Originally posted 2020-09-21 19:50:46.406000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>I&#039;m going to make a reduction of Lipiny and Raszczyce, which, as Friesach, I use more often. Then I&#039;m considering to do the same with Cracow and Giubiasco, which seens to be more popular.
Have a nice study!

**Comment by larspalo** [comment]
Originally posted 2020-09-21 10:49:54.610000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>I downloaded the Freisach light organ from the link above and there were many problems. I am using Windows - but that should make no difference.

1. The windows unzipper would not unzip the file, but 7zip did appear to work OK - although it took a long time.
2. When loading the sampleset into GO there were thousands of errors shown in the log file and it never did load. I had to use the task manager to close GO as it appeared to be locked up and unresponsive.

Has anyone else tried this link?

csw900

**Reply by larspalo** [reply]
Originally posted 2020-09-21 20:01:40.120000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>I&#039;m sorry for that.
I&#039;m assuming you put Only the folder named as &quot;Data - Friesach&quot; in the GO folder called &quot;Organs&quot;. Otherwise the .organ file could not find the sample folder.

It&#039;s necessarry to set &quot;Release loading&quot; to &quot;Single release&quot; in &quot;Audio/Midi Settings&quot;, as I wrote above, so I think you did it.

If was not any of these problems, may the uncompressed folder be corrupted for some reason. In this case, you can try to download and unzip again. 
I&#039;m doing right now another compression in other format and I&#039;ll upload it somewhere.

All the best!

**Reply by larspalo** [reply]
Originally posted 2020-09-22 02:28:57.802000 by colodetti in GrandOrgue discussion forum on SourceForge<br><br>A link for another Friesach light: https://mega.nz/file/KhMnlAjL#LtflRHDLf--nEBXhBujjq9RCDT_30cSKQWCvqqop-mY

**Reply by larspalo** [reply]
Originally posted 2020-09-22 06:30:46.130000 by marcofrancesco in GrandOrgue discussion forum on SourceForge<br><br>Your uploaded file is ok. I downloaded the first one and after editing 
the ODF according to my folder name (I renamed your new folder &quot;Data - 
Friesach (LIGHT)&quot;) it loaded well in GO.

Regards
Mark

On 21/09/2020 22:01, Diego Colodetti wrote:
&gt;
&gt; I&#039;m sorry for that.
&gt; I&#039;m assuming you put Only the folder named as &quot;Data - Friesach&quot; in the 
&gt; GO folder called &quot;Organs&quot;. Otherwise the .organ file could not find 
&gt; the sample folder.
&gt;
&gt; It&#039;s necessarry to set &quot;Release loading&quot; to &quot;Single release&quot; in 
&gt; &quot;Audio/Midi Settings&quot;, as I wrote above, so I think you did it.
&gt;
&gt; If was not any of these problems, may the uncompressed folder be 
&gt; corrupted for some reason. In this case, you can try to download and 
&gt; unzip again.
&gt; I&#039;m doing right now another compression in other format and I&#039;ll 
&gt; upload it somewhere.
&gt;
&gt; All the best!
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Install GrandOrgue on Raspberry Pi 4 8GB under RPiOS 64 bit 
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962124/thread/7daeed0824/?limit=25#0572/fa69&gt;
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Sent from sourceforge.net because you indicated interest in 
&gt; https://sourceforge.net/p/ourorgan/discussion/962124/ 
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962124/&gt;
&gt;
&gt; To unsubscribe from further messages, please visit 
&gt; https://sourceforge.net/auth/subscriptions/ 
&gt; &lt;https://sourceforge.net/auth/subscriptions/&gt;
&gt;



-- 
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus
<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/7daeed0824/0572/fa69/b418/attachment/alternate<br>

**Reply by larspalo** [reply]
Originally posted 2020-09-22 13:18:51.055000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>You were right, the problem was that the loader could not find the sample data. However GO crashed when next I loaded the organ. I only have 8GB of memory so it might be a bit marginal for reliability.

Anyway the manual graphic is too small for my liking so I will be deleting this organ.

However it was useful to try it out, Thanks.

csw900

**Comment by larspalo** [comment]
Originally posted 2020-09-25 23:15:43.801000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>Raspian 10 on OBS only has support for 32 bits (armv7l). 
GO offers ARM 64 bit binaries (aarch64) for other distributions. See https://build.opensuse.org/package/show/home:e9925248:grandorgue/grandorgue

**Comment by larspalo** [comment]
Originally posted 2020-11-08 17:54:20.533000 by jrc in GrandOrgue discussion forum on SourceForge<br><br>can someone please enlighten me as to how I install GO from the zip compilation Erik DS kindly offers? I have unzipped the file, to get 3 files, but I don&#039;t know whether to seperately install each, or whether I need some command to &#039;blend them&#039; together!
Help!!!

**Reply by larspalo** [reply]
Originally posted 2020-11-08 19:09:07.207000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>The 3 files are .deb files. They are intended to be installed one by one 
under the 64 bit Operating System of Raspberry Pi.
The file grandorgue-demo is optional; you can use some of the available 
sample sets provided they fit in the memory you have available.

All the best.

---------------------------------------------

john childs wrote:
&gt;
&gt; can someone please enlighten me as to how I install GO from the zip 
&gt; compilation Erik DS kindly offers? I have unzipped the file, to get 3 
&gt; files, but I don&#039;t know whether to seperately install each, or whether 
&gt; I need some command to &#039;blend them&#039; together!
&gt; Help!!!
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Install GrandOrgue on Raspberry Pi 4 8GB under RPiOS 64 bit 
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962124/thread/7daeed0824/?limit=25#e055&gt;
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Sent from sourceforge.net because you indicated interest in 
&gt; https://sourceforge.net/p/ourorgan/discussion/962124/
&gt;
&gt; To unsubscribe from further messages, please visit 
&gt; https://sourceforge.net/auth/subscriptions/
&gt;

**Comment by larspalo** [comment]
Originally posted 2021-01-01 11:19:08.222000 by jopiek in GrandOrgue discussion forum on SourceForge<br><br>I can install the demo and resources files but grandorgue says: &quot;a package could not be found that allows the action to complete&quot;. The detailed report it promises mus be somewhere in some log file but I don&#039;t know where yet.

**Update:**
I just used the script https://github.com/djbottrill/GrandOrguePi to download and compile sources. Works like a charm on Pi4 8Gb with 64-bit PiOS (Raspbian).

**Comment by larspalo** [comment]
Originally posted 2021-01-01 14:43:58.899000 by erikds in GrandOrgue discussion forum on SourceForge<br><br>Under RPiOS64 i have following files installed:

libwxgtk3.0-0v5
libwxgtk3.0-gtk3-0v5
libwxgtk3.0-gtk3-dev

Possibly you do not need the last file for runtime.
I needed it at compile time.

If these files are not installed you can install them using:
Preferences- Add/Remove software.

I hope this helps.

Best regards.

**Comment by larspalo** [comment]
Originally posted 2021-01-01 15:19:34.159000 by jopiek in GrandOrgue discussion forum on SourceForge<br><br>Thanks, it complained so I installed them too indeed. Works like a charm.

**Comment by lbssousa** [comment]
arm64 packages for Raspberry Pi OS 64-bit are already available here: https://github.com/GrandOrgue/grandorgue/releases I'm currently using these ones.
